File: Python-Examples/Basics/Blink.py
"""
Blink.py
This example makes a LED blink every second. It uses the Python programming language
and the MRAA library.

This example code is in the public domain.

Revision History
------------------------------------------------
Author			  Date			  Description
------------------------------------------------
Carlos Mata			1-22-2015		Example created
"""

import mraa #calls the MRAA library
import time #calls the time library

#Setup - Allows to choose the pin 13 (Built-in LED) as an output.

x = mraa.Gpio(13)
x.dir(mraa.DIR_OUT)

#Process - The process is going to change the state of the pin every second

try:
	print "It's blinking"
	while True:
		x.write(1)
		time.sleep(1)  #Maintain the status of the pin for a second
		x.write(0)
		time.sleep(1)
except KeyboardInterrupt:
        print ""

File: JavaScript-Examples/Blink.js
//Example Blink.js
//
//This examples shows how to make a simple Blink script
//
//This example code is in the public domain.
//
//Revision History
//------------------------------------------------
//Author			  Date			  Description
//------------------------------------------------
//Pedro Mora		02-12-2015			Blink.js
//
//

var mraa = require('mraa');                                   //this script requires MRAA
console.log('MRAA Version: ' + m.getVersion());               //this will write the MRAA version
var BLED = new m.Gpio(13);                                    //this binds the variable BLED to pin 13
BLED.dir(mraa.DIR_OUT);                                       //this sets the GPIO to output
var LEDSt = true;                                             //Boolean variable that will remember state of Led
blink();                                                      //this will call the blink function
function blink()
{
	BLED.write(LEDSt?1:0);                                //if the state of the LED is true then write a '1' (high) otherwise it writes a '0' (low)
	LEDSt = !LEDSt;                                       //this will invert the state of the LED
	setTimeout(blink,1000);                               //call the indicated function after 1 second(1000 milliseconds)
}
